nightfall-client/src/services/transaction.mjs

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable import/prefer-default-export */
+
+import axios from 'axios';
+import logger from '@polygon-nightfall/common-files/utils/logger.mjs';
+import getProposers from '@polygon-nightfall/common-files/utils/proposer.mjs';
+import NotFoundError from '@polygon-nightfall/common-files/utils/not-found-error.mjs';
+import ValidationError from '@polygon-nightfall/common-files/utils/validation-error.mjs';
+
+const STATUS_MINED = 'mined';
+const STATUS_MEMPOOL = 'mempool';
+
+export async function findTransactionInMempools(l2TransactionHash) {
+  logger.debug('Get all registered proposer URLs from State contract..');
+  const proposerURLs = (await getProposers()).map(p => p.url);
+
+  logger.debug('Query each proposer mempool..');
+  const promises = [];
```

Comment:
```js
const promises = proposerURLs.map((pURL) =>
  axios.get(`${pURL}/proposer/mempool/${l2TransactionHash}`)
);
```

Reply:
And we cam scrap forEach.

Reply:
I agree. It's cleaner with map.

Reply:
Thanks! Very clever, I'm actually annoyed that it didn't occur to me :nerd_face: I really enjoy array methods, must have a sad life...